search/materialize_test.go

```diff
@@ -178,7 +180,149 @@ func query(t *testing.T, mint, maxt int64, shard storage.ParquetShard, constrain
 		require.NoError(t, err)
 		series, err := m.Materialize(ctx, i, mint, maxt, false, rr)
 		require.NoError(t, err)
-		found = append(found, series...)
+		found = append(found, pullSeries(t, series)...)
+	}
+	return found
+}
+
+func pullChunks(t *testing.T, chnks prom_storage.ChunkSeries) []chunks.Meta {
+	it := chnks.Iterator(nil)
+	var metas []chunks.Meta
+	for it.Next() {
+		metas = append(metas, it.At())
+	}
+	if err := it.Err(); err != nil {
+		t.Fatalf("error iterating chunks: %v", err)
+	}
+	return metas
+}
+
+func pullSeries(t *testing.T, series prom_storage.ChunkSeriesSet) []prom_storage.ChunkSeries {
+	var found []prom_storage.ChunkSeries
+	for series.Next() {
+		found = append(found, series.At())
+	}
+	if err := series.Err(); err != nil {
+		t.Fatalf("error iterating series: %v", err)
 	}
 	return found
 }
+
+func BenchmarkMaterialize(b *testing.B) {
+	ctx := context.Background()
+	st := teststorage.New(b)
+	b.Cleanup(func() { _ = st.Close() })
+
+	bkt, err := filesystem.NewBucket(b.TempDir())
```

Comment:
Maybe we can simulate some latency here? 5ms per call?

Reply:
Good idea! I bet this will show the impact of dropping concurrency, I missed this. As I said in my other comment, I will take a look at this and report back